packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.tsx

```diff
@@ -60,50 +82,53 @@ export class Checkbox extends BaseComponent<ICheckboxProps, ICheckboxState> impl
       inputProps,
       label,
       name,
-      boxSide
+      boxSide,
+      theme,
+      styles: customStyles,
         } = this.props;
 
-    const { isFocused } = this.state;
     const isChecked = checked === undefined ? this.state.isChecked : checked;
+    const isReversed = boxSide !== 'start' ? true : false;
 
-    let rootClassName = css('ms-Checkbox', className, styles.root, {
-      ['reversed ' + styles.reversed]: boxSide !== 'start',
-      ['is-checked ' + styles.isChecked]: isChecked,
-      ['is-disabled ' + styles.isDisabled]: disabled,
-      ['is-inFocus ' + styles.isInFocus]: isFocused
-    });
+    const classNames = this._getClassNames(
+      getStyles(theme, customStyles),
+      className,
+      disabled,
+      isChecked,
+      isReversed
+    );
 
     return (
-      <div className={ rootClassName } style={ this.props.style } >
-        <input
-          { ...inputProps }
-          { ...(checked !== undefined && { checked }) }
-          { ...(defaultChecked !== undefined && { defaultChecked }) }
-          disabled={ disabled }
-          ref={ this._resolveRef('_checkBox') }
-          id={ this._id }
-          name={ name || this._id }
-          className={ css('ms-Checkbox-input', styles.input) }
-          type='checkbox'
-          onChange={ this._onChange }
-          onFocus={ this._onFocus }
-          onBlur={ this._onBlur }
-          aria-checked={ isChecked }
-        />
-        <label className={ css('ms-Checkbox-label', styles.label) } htmlFor={ this._id }>
-          <span className={ css('ms-Checkbox-box', styles.box) }>
-            <span className={ css('ms-Checkbox-checkbox', styles.checkbox) }>
-              <Icon iconName='CheckMark' className={ styles.checkmark } />
-            </span>
-          </span>
-          { label && <span className={ css('text', styles.text) }>{ label }</span> }
+      <button
```

Comment:
Yes, but there is already a nonzero chance of regression by converting the styles as is to glamor. By making additional changes, that increases that risk and makes it more difficult for us to figure out if new issues are from the glamor port or from the "clean up". I don't think these issues should block this PR and should be done in a follow up PR.